nexus/types/src/external_api/params.rs

```diff
@@ -83,6 +89,44 @@ pub struct OptionalProjectSelector {
     pub project_selector: Option<ProjectSelector>,
 }
 
+#[derive(Deserialize, JsonSchema)]
+pub struct DiskSelector {
+    #[serde(flatten)]
+    pub project_selector: Option<ProjectSelector>,
```

Comment:
do we prefer this style to 
```suggestion
    pub project: Option<NameOrId>,
    pub organization: Option<NameOrId>,
```

Does it allow for additional code sharing for these nested, flattened selector structures?

Reply:
By encoding a selector at every level (which itself is a resource + possible parent selector) we can flatten both the creation of selectors and the lookup functions down to only concerning itself with two cases. If you look at the endpoint handlers they're always using using a selector and potentially a path param. My thought here is that this general shape will be easier to encode in a macro. Once organization goes away and project is flattened a bit of the structure can be reduced though.